src/main/java/stsjorbsmod/patches/BurningTimeEaterPatch.java

```diff
@@ -0,0 +1,56 @@
+package stsjorbsmod.patches;
+
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.animations.TalkAction;
+import com.megacrit.cardcrawl.actions.common.HealAction;
+import com.megacrit.cardcrawl.actions.unique.RemoveDebuffsAction;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.beyond.TimeEater;
+import com.megacrit.cardcrawl.vfx.CloudBubble;
+import com.megacrit.cardcrawl.vfx.SpeechBubble;
+import com.megacrit.cardcrawl.vfx.ThoughtBubble;
+import javassist.CannotCompileException;
+import javassist.expr.ExprEditor;
+import javassist.expr.MethodCall;
+import stsjorbsmod.JorbsMod;
+import stsjorbsmod.powers.BurningPower;
+
+import java.util.Iterator;
+
+public class BurningTimeEaterPatch {
+    private static final String UI_ID = JorbsMod.makeID(BurningTimeEaterPatch.class);
+    private static final String[] TEXT = CardCrawlGame.languagePack.getUIString(UI_ID).TEXT;
+
+    @SpirePatch(clz = RemoveDebuffsAction.class, method = "update")
+    public static class RemoveDebuffsAction_Update_BurningTimeEater {
```

Comment:
I think you could probably simplify and avoid a patch for this by instead adding something like this to `BurningPower`:

```java
    @Override
    public void onRemove() {
        // This makes the heal prevention affect Time Eater's "cleanse and heal simultaneously" turn
        // despite the cleanse technically happening first.
        for (AbstractGameAction a : AbstractDungeon.actionManager.actions) {
            if (TimeEater.ID.equals(this.owner.id) && a instanceof HealAction && a.target == this.owner) {
                a.amount = this.onHeal(a.amount);
            }
        }
    }
```

Reply:
Oh yeah... forgot to check the overrideable methods on AbstractPower... Yeah I'll take a look and see.
Another thing I was considering was to simply remove the healing action all together. Heal for 0 doesn't have the animation as far as I can tell.